partition_snapshot_reader.hh

```diff
@@ -165,6 +203,7 @@ class partition_snapshot_flat_reader : public flat_mutation_reader::impl, public
                                       logalloc::region& region, logalloc::allocating_section& read_section,
                                       bool digest_requested)
             : _schema(s)
+            , _snapshot_schema(Reversing ? s.make_reversed() : s.shared_from_this())
```

Comment:
Again, if we have two schemas, both their names should reflect their roles (e.g. _query_schema and _snapshot_schema).